src/Middleware/Microsoft.AspNetCore.OutputCaching.StackExchangeRedis/src/RedisCacheOptions.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using StackExchange.Redis;
+using StackExchange.Redis.Configuration;
+using StackExchange.Redis.Profiling;
+
+namespace Microsoft.AspNetCore.OutputCaching.StackExchangeRedis;
+
+/// <summary>
+/// Configuration options for Redis based output cache.
+/// </summary>
+public class RedisOutputCacheOptions : IOptions<RedisOutputCacheOptions>
```

Comment:
```suggestion
public class RedisOutputCacheOptions
```

This class doesn't need to implement IOptions. I know the existing distributed cache implements it, but I believe that is a mistake (that we can't fix because it is a shipped API).

Option classes are POCOs, and when someone consumes them in DI, that's when they use `IOptions<XXX>`.

For example:

https://github.com/dotnet/aspnetcore/blob/00d0038f937f0059a847fde504649fe33ec935e0/src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs#L36-L37

https://github.com/dotnet/aspnetcore/blob/00d0038f937f0059a847fde504649fe33ec935e0/src/Servers/Kestrel/Core/src/KestrelServerOptions.cs#L27-L28

`KestrelServerOptions` doesn't implement IOptions itself.

Reply:
We did it in a few types like MemoryCache/Options where people would actually new up the options and class themselves. That's likely not the case here.

Reply:
Remove it. You don't new up a RedisOutputCache instance directly do you?

Reply:
no; nuking, ta

Reply:
ah, it was probably there for the test; I switched them to use DI properly instead - sound like a plan? https://github.com/dotnet/aspnetcore/pull/50457/commits/7c915d43b5fdd10c676a2813c963bf681a07391c#diff-9b6a4cced77cfd298a082285b4f112b7e5a97059a80e5291df832b037e0e996a

Reply:
Great.